### DISCUSSION TITLE
Tailwind fails to compile animate-pulse: "Cannot set property 'parent' of undefined."

### DISCUSSION THREAD

**Discussion by nataliemeurer** [tailwindlabs/tailwindcss#8853] [Help]
Hi all,

My team is encountering an issue building tailwind for specific classes locally, particularly `.animate-pulse`. Here is some information on our version, setup, and error.

## Versions
    "postcss": "8.4.14",
    "tailwindcss": "^3.1.6"

## Build System

We are running a create-react-app setup with craco for configuration. We are using postcss with the following craco configuration:

```
const appDirectory = fs.realpathSync(process.cwd());
const resolveApp = (relativePath) => path.resolve(appDirectory, relativePath);

module.exports = {
	style: {
		postcss: {
      plugins: [
        require('tailwindcss'),
        require('autoprefixer'),
      ],
    }
	},
	plugins: [
		{
			plugin: require('craco-babel-loader'),
			options: {
				includes: [resolveApp('../api')],
			},
		},
	],
};
```

## Error
```
$ tailwind build -i src/styles/index.css -o src/styles/tailwind.css

🌼 daisyUI components 2.13.6 
  ✔︎ Including:  base, components, themes[1]
  
node:internal/process/promises:246
          triggerUncaughtException(err, true /* fromPromise */);
          ^

TypeError: Cannot set property 'parent' of undefined
    at Root.removeChild (my/project/node_modules/postcss/lib/container.js:522:30)
    at Root.removeChild (my/project/node_modules/postcss/lib/root.js:49:45)
    at Proxy.remove (my/project/packages/react-app/node_modules/postcss/lib/node.js:78:19)
    at Rule (my/project/packages/react-app/node_modules/postcss-nested/index.js:209:43)
    at LazyResult.visitSync (my/project/packages/react-app/node_modules/postcss/lib/lazy-result.js:308:19)
    at LazyResult.walkSync (my/project/packages/react-app/node_modules/postcss/lib/lazy-result.js:297:20)
    at my/project/packages/react-app/node_modules/postcss/lib/lazy-result.js:291:39
    at Root.each (my/project/node_modules/postcss/lib/container.js:105:16)
    at LazyResult.walkSync (my/project/packages/react-app/node_modules/postcss/lib/lazy-result.js:290:16)
    at LazyResult.sync (my/project/packages/react-app/node_modules/postcss/lib/lazy-result.js:247:14) {
  postcssNode: <ref *3> Rule {
    raws: { semicolon: true },
    selector: '.animate-pulse',
    type: 'rule',
    nodes: [],
    parent: <ref *1> Root {
      raws: {},
      type: 'root',
      nodes: [
        <ref *2> AtRule {
          raws: {},
          name: 'keyframes',
          params: 'pulse',
          type: 'atrule',
          nodes: [ [Rule] ],
          parent: [Circular *1],
          proxyCache: [Circular *2],
          lastEach: 1,
          indexes: {},
          [Symbol(my)]: true
        },
        [Circular *3]
      ],
      lastEach: 1,
      indexes: { '1': 1 },
      [Symbol(my)]: true,
      [Symbol(isClean)]: false
    },
    proxyCache: [Circular *3],
    lastEach: 1,
    indexes: {},
    [Symbol(my)]: true,
    [Symbol(isClean)]: false
  }
}
error Command failed with exit code 1.
```

In other words, tailwind is working great for the majority of classes, but it's consistently failing if we add `animate-pulse` or other animations to our components.

Any help would be greatly appreciated. Happy to dig in here if folks have a sense of what the issue is.

**Comment by arisac** [comment]
Can you create a demo repo for reproduction for your issue?

**Reply by nataliemeurer** [reply]
Sure I can probably do that!

**Comment by nataliemeurer** [comment]
Hi all,

We resolved this over on our end. The issue was conflicting versions of tailwind at the root of our yarn workspace and our react-app. We resolved to only use one version in our react-app, then fixed up some postcss issues and ended up in a good state.

In case it helps someone in the future, here's our new craco config as well as primary versions:


```
// versions
"postcss": "^8.4.14",
"tailwindcss": "^3.1.6"
"react-scripts": "^5.0.1",
"@craco/craco": "^6.4.3",
```

```
// craco.config.js
module.exports = {
	eslint: {
		enable: true,
  },
	style: {
		postcssOptions: {
      plugins: [
        require('tailwindcss'),
        require('autoprefixer'),
      ],
    }
	},
	plugins: [
		{
			plugin: require('craco-babel-loader'),
			options: {
				includes: [resolveApp('../api')],
				excludes: [resolveApp('./src/assets')]
			},
		},
	],
};
```

**Reply by arisac** [reply]
Glad you fixed it and thanks for sharing.

**Comment by arisac** [comment]
@nataliemeurer  please close this discussion and mark as answered so other people don't need come in to see if you still help.
